src/main/java/com/google/sps/servlets/CommentService.java

```
@@ -0,0 +1,77 @@
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.api.services.youtube.YouTube;
+import java.io.IOException;
+import java.lang.Exception;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+
+public class CommentService {
+
+  private final String DEVELOPER_KEY = "DEV_KEY";
+  private static final String APPLICATION_NAME = "YouTube Comments";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  private YouTube getService() throws GeneralSecurityException, IOException{
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    return new YouTube.Builder(httpTransport, JSON_FACTORY, null)
+        .setApplicationName(APPLICATION_NAME)
+        .build();
+  }
+
+  /**
+   * Calls YouTube Data API to return a list of the top 25 comments for a video, identified by its
+   * video id. 
+   */
+  public List<String> getCommentsFromId(String videoId)
+      throws GeneralSecurityException, IOException, GoogleJsonResponseException {
+    YouTube youtubeService = getService();
+    YouTube.CommentThreads.List request = youtubeService.commentThreads()
+        .list(Collections.singletonList("snippet"));
+    try { 
+      // Call YouTube API and get the 25 comments that appear first in comment section of the video
+      CommentThreadListResponse response = request.setKey(DEVELOPER_KEY)
+          .setMaxResults(25L)
+          .setOrder("relevance")
+          .setTextFormat("plainText")
+          .setVideoId(videoId)
+          .execute();
+      
+      if (response.getItems().isEmpty()) {
+        throw new Exception(String.format("Video %s has no comments.", videoId));
```

Why did you decide to throw an exception instead of returning an empty list?

---

Yeah I think throwing an exception was pretty pointless now that I think about it... I guess I could write tests to check that the correct exception was thrown in different cases?